cumulusci/core/config/project_config.py

```diff
@@ -498,8 +509,76 @@ def get_repo_from_url(self, url):
 
         return repo
 
-    def get_ref_for_dependency(self, repo, dependency, include_beta=None):
-        release = None
+    def find_matching_2gp_release(
+        self, remote_repo: Union[BaseTaskFlowConfig, Mock]
+    ) -> Tuple[str, str]:
+        # To allow us to locate release branches on the remote repo, we need to know its feature branch prefix.
+        # We'll use the cumulusci.yml file from HEAD on the main branch to determine this.
+
+        release_id = get_release_identifier(
+            self.repo_branch, self.project__git__prefix_feature
+        )
+        remote_branch_prefix = None
+        contents = remote_repo.file_contents(
+            "cumulusci.yml",
+            ref=remote_repo.branch(remote_repo.default_branch).commit.sha,
+        )
+        head_cumulusci_yml = cci_safe_load(
+            io.StringIO(contents.decoded.decode("utf-8"))
+        )
+        remote_branch_prefix = (
+            head_cumulusci_yml.get("project", {})
+            .get("git", {})
+            .get("prefix_feature", "feature/")
+        )
+
+        remote_matching_branch = construct_release_branch(
+            remote_branch_prefix, release_id
+        )
+
+        # Check the most recent five commits on this release branch looking for a 2GP package to use
+        try:
+            release_branch = remote_repo.branch(remote_matching_branch)
+        except NotFoundError:
+            self.logger.info(
+                f"Release branch {remote_matching_branch} not found on {remote_repo.clone_url}. Falling back to 1GP."
+            )
+            return (None, None)
+
+        version_id = None
+        count = 0
+        commit = release_branch.commit
+        while version_id is None and count < 5:
```

Comment:
You might be able to use fewer github API requests by using https://github3py.readthedocs.io/en/master/api-reference/repos.html#github3.repos.repo.Repository.commits rather than getting one commit at a time and following the parent references yourself.